### DISCUSSION TITLE
uploading new image or file to the file system doesn't show after `next build`

### DISCUSSION THREAD

**Discussion by Robinzon100** [vercel/next.js#13351] [Help] [answered]
when I sen a file and text to the API, it saves the file in the file system (like in: `public/news`) and takes it's file path and stores it to the DB (like this: `/uploads/news/(1590330102136)--anykey.png`) and the I just render this element:

```javascript
    <div
        className="news_card__image"
        style={{
            backgroundImage: `url(" ${path.join(__dirname, props.imageUrl)}")`
        }}
        alt={props.title}>
    </div>
```
**BUT**
after i do a `next build` and upload a new file with text; it renders out the text but the image is giving me a 404 error, despite it having the exact same path as other images that were uploaded before the `next build` command

**Comment by jamesmosier** [comment, marked as answer]
You shouldn't be using `path` in your client rendered application. Your user's web browser will not know about your server's local `path` at the time of rendering.



Duplicate of https://github.com/zeit/next.js/discussions/13321

**Reply by Robinzon100** [reply]
sorry this was an early attempt, i was trying to make it work but it didn't, the code really looks like this
```javascript
    <div
        className="news_card__image"
        style={{
            backgroundImage: `url(" ${props.imageUrl}")`
        }}
        alt={props.title}>
    </div>
```
ant the path this brings looks like this `/uploads/news/(1590330102136)--anykey.png`

P.S: the files are being saved in the file system

thank you for the reply btw : @jamesmosier

**Reply by Robinzon100** [reply]
and in the network tab, I see that the image is giving me a 404 error

**Reply by jamesmosier** [reply]
After your site is built, you cannot dynamically add more images to the `public` folder and have them accessible to the end user. The `public` folder isn't really meant for that, rather it is just meant for the developer to store static assets to be served.

You'll want to consider saving the images in a CDN or otherwise you'll want to save them somewhere on your server that you can make available via your proxy configuration.

**Reply by Robinzon100** [reply]
thank you @jamesmosier, appreciate it
if you have any resources I can follow I will extremely appreciate it,
have a beautiful day ☀

**Reply by jamesmosier** [reply]
There are a bunch of solutions out there depending on where you are hosting as well as your server setup. It's hard to have a 1 size fits all solution for this if you wanted to just upload them to your server.

If you wanted a managed solution, [Cloudinary](https://cloudinary.com) might be a good place to start as they have a decent free plan and you can just upload images there programmatically. They also have a [React uploader](https://cloudinary.com/documentation/react_image_and_video_upload) you could even consider otherwise you could use your API to upload your images. 

Aside from that, [imgix](https://www.imgix.com/), AWS S3, Google Cloud, etc. all have static file hosting solutions.

**Reply by Robinzon100** [reply]
thank you

**Reply by MhZmn** [reply]
Is there a way to host images locally on VPS deployment? @jamesmosier
